### PR TITLE
Make errorCode of InsertTimeOutOfBoundsFault consistent with others

### PR DESCRIPTION
**Pull request by zachjsh**
The errorCode of this fault when serialized over the wire was being set to the name of the class `InsertTimeOutOfBoundsFault` instead of the CODE `InsertTimeOutOfBounds`. All other faults' errorCodes are serialized as the respective Fault's code, so making consistent here as well. I checked all existing faults at the moment, and this was the only one I found to be inconsistent.

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by zachjsh** [comment]
> Nice catch, thanks @zachjsh.

https://tenor.com/view/thanks-american-psycho-christian-bale-gif-13090772

**Comment by zachjsh** [comment]
> Nice catch, thanks @zachjsh.

image

**Comment by zachjsh** [comment]
> Nice catch, thanks @zachjsh.

image

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/error/InsertTimeOutOfBoundsFault.java [hunk 1]
```diff
@@ -20,10 +20,12 @@
 package org.apache.druid.msq.indexing.error;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
 import org.joda.time.Interval;
 
 import java.util.Objects;
 
+@JsonTypeName(InsertTimeOutOfBoundsFault.CODE)
```
Comment (abhishekrb19) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/error/InsertTimeOutOfBoundsFault.java:28]: Do we have any unit test that tests the serialization/deserialization of this class's wire format?

Comment (zachjsh) [reply extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/error/InsertTimeOutOfBoundsFault.java:28]: yes in org.apache.druid.msq.indexing.error.MSQFaultSerdeTest

**Pull request merged**